app/src/test/java/org/oppia/app/player/state/StateFragmentLocalTest.kt

```diff
@@ -246,6 +255,73 @@ class StateFragmentLocalTest {
     }
   }
 
+  @Test
+  @InternalCoroutinesApi
+  @ExperimentalCoroutinesApi
+  fun testStateFragment_submitTwoWrongAnswers_checkPreviousHeaderVisible() {
+    launchForExploration(FRACTIONS_EXPLORATION_ID_1).use {
+      startPlayingExploration()
+      playThroughState1()
+
+      submitTwoWrongAnswers()
+
+      onView(withId(R.id.previous_response_header)).check(matches(isDisplayed()))
+    }
+  }
+
+  @Test
+  @InternalCoroutinesApi
+  @ExperimentalCoroutinesApi
+  fun testStateFragment_submitTwoWrongAnswers_checkPreviousHeaderCollapsed() {
+    launchForExploration(FRACTIONS_EXPLORATION_ID_1).use {
+      startPlayingExploration()
+      playThroughState1()
+
+      submitTwoWrongAnswers()
+
+      onView(withId(R.id.previous_response_header)).check(matches(isDisplayed()))
+      onView(withId(R.id.state_recycler_view)).check(matches(hasChildCount(/* childCount= */ 5)))
+    }
+  }
+
+  @Test
+  @InternalCoroutinesApi
+  @ExperimentalCoroutinesApi
+  fun testStateFragment_submitTwoWrongAnswers_expandPreviousResponse_checkPreviousHeaderExpanded() {
+    launchForExploration(FRACTIONS_EXPLORATION_ID_1).use {
+      startPlayingExploration()
+      playThroughState1()
+
+      submitTwoWrongAnswers()
+
+      onView(withId(R.id.previous_response_header)).check(matches(isDisplayed()))
+      onView(withId(R.id.state_recycler_view)).perform(scrollToViewType(PREVIOUS_RESPONSES_HEADER))
+      onView(withId(R.id.previous_response_header)).perform(click())
+      onView(withId(R.id.state_recycler_view)).check(matches(hasChildCount(/* childCount= */ 7)))
+    }
+  }
+
+  @Test
+  @InternalCoroutinesApi
+  @ExperimentalCoroutinesApi
+  fun testStateFragment_submitTwoWrongAnswers_expandPreviousResponse_collapsePreviousResponse_checkPreviousHeaderCollapsed() { // ktlint-disable max-line-length
```

Comment:
Please abbreviate as needed to fit this--prefer not disabling the max line-length ktlint check (or any ktlint checks for that matter).

Reply:
> Thanks @anandwana001. Do the new tests actually fail if the ObservableArrayList change is reverted? If not, can you add new question player-dedicated tests for this purpose? We should make sure this case is covered by tests.

Yes, we do need something specific for Question player. 
Currently, `StateFragmentTestActivity` is currently built according to exploration. 
If we plan to add Question specific tests in `StateFragmentLocalTest` Do you recommend to make changes in it, in such a way so that we can either trigger to start an exploration or we should create a complete system like `StateFragmentTestActivity` separately for Question Player?

Reply:
The latter. We should have a separate test suite for questions.

Reply:
> Please abbreviate as needed to fit this--prefer not disabling the max line-length ktlint check (or any ktlint checks for that matter).

Done